### PR TITLE
Add new tests to improve coverage.

### PR DESCRIPTION
**Pull request by luo-yuyuan**
#348 I added tests to 5 modules currently with no tests.

### CODE REVIEW

**Comment by Androbin** [comment]
Too bad, some of these modules are currently being removed by #441.

**Comment by sushain97** [comment]
Good start, @luo-yuyuan ! I haven't looked at the code yet but it appears that the % of code coverage hasn't actually increased? See L732 on https://travis-ci.org/apertium/phenny/jobs/446002521 versus master.

**Comment by Androbin** [comment]
@sushain97 That's was to be expected. Coverage is only computed over the set of modules referenced by tests.
https://www.diffchecker.com/h5bofXFj

**Comment by Androbin** [comment]
However I have no idea why coverage for `metar.py` decreased?

**Comment by Androbin** [comment]
`after_success: coveralls` is executed after build and thus doesn't fail the build in case of low coverage.
If you want to change how coverage is calculated, you can still do so with the existing `nosetests` setup.

**Comment by luo-yuyuan** [comment]
@Androbin How would I do that? I couldn't find the setup file for nosetests... Is there one? If so, which is it, and if not, how would I look at the settings? Thanks!

**Comment by Androbin** [comment]
That would be `unit-tests.sh`. You can use the glob `modules/*.py` or similar to pass to the 
`nosetests` command.

**Comment by luo-yuyuan** [comment]
Ok cool, I have just changed the coverage badge in readme from linking to my forked repo to linking to this one. What else should I do in order to complete this GCI task? Thanks :)

**Comment by wei2912** [comment]
Some brief remarks:
1. `test_choose.py` should try testing with more than two items, as well as cases with zero or one item(s).

File: modules/test/test_apertiumchill.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class TestApertiumchill(unittest.TestCase):
+    def setUp(self):
+        self.phenny = MagicMock()
+        self.input = MagicMock()
+
+    def test_chill(self):
+        apertiumchill.chill(self.phenny, self.input)
+        self.assertTrue(self.phenny.say.called)
```
Comment (sushain97) [inline modules/test/test_apertiumchill.py]: Please add newlines to the end of all files.

Comment (sushain97) [reply modules/test/test_apertiumchill.py]: This test doesn't actually test very much? :\

File: modules/test/test_botfun.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import unittest
+from mock import MagicMock
+from modules import botfun
+
+
+class TestChoose(unittest.TestCase):
```
Comment (sushain97) [inline modules/test/test_botfun.py]: Is this the right name for this class?

File: modules/test/test_botfun.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        messages = ["hits ChanServ", "punches ChanServ", "kicks ChanServ", "hits ChanServ with a rubber hose",
+                    "stabs ChanServ with a clean kitchen knife"]
+        botfun.botfight(self.phenny, self.input)
+        out = self.phenny.do.call_args[0][0]
+        self.assertTrue(out in messages)
+
+    def test_bothug(self):
+        botfun.bothug(self.phenny, self.input)
+        out = self.phenny.do.call_args[0][0]
+        self.assertTrue(out == "hugs ChanServ")
```
Comment (sushain97) [inline modules/test/test_botfun.py]: Seems like you can use `assert_called_once_with` or something of the sort here?

File: modules/test/test_choose.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    def setUp(self):
+        self.phenny = MagicMock()
+        self.input = MagicMock()
+
+    def test_valid(self):
+        self.input.group = lambda x: ['.choose', 'canada usa'][x]
+        choose.choose(self.phenny, self.input)
+        out = self.phenny.reply.call_args[0][0]
+        m = (out == 'canada' or out == 'usa')
+        self.assertTrue(m)
```
Comment (sushain97) [inline modules/test/test_choose.py]: Similarly below:

```suggestion
        self.assertTrue(out == 'canada' or out == 'usa')
```

**Comment by sushain97** [comment]
@wei2912 I don't really mind the changes to coverage also being here but if @luo-yuyuan is willing to cherry-pick them out, that's fine.

I do agree with the comments re. `test_choose` testing.

**Comment by Androbin** [comment]
@sushain97 I've fixed the coverage report in #443.
I have also tested this with the changes by @luo-yuyuan.
Here's a diff of the coverage without vs. with the new tests:
https://www.diffchecker.com/CAalTp1H

**Comment by sushain97** [comment]
Ah, thanks for the diff @Androbin ! @luo-yuyuan - looks like a few more percentage could be warranted here? Since 10% of the full codebase is a bit more than I expected, I'll modify the task to be 5%. Feel free to hit like 4% or something since you also added the coverage bits.

**Comment by Androbin** [comment]
Keep in mind that even modules without dedicated tests get some coverage via other modules.
I suggest looking at the stats and focusing on modules with many lines and low coverage.

For example `git.py` has 376 lines of which 326 are untested.
(For reference, you added coverage to 74 lines till now)

**Comment by Androbin** [comment]
One percentage point is about 54 lines. `git.py` alone is potentially worth 6% coverage.

**Comment by Androbin** [comment]
After you're done, please increase the coverage threshold in `unit-tests.sh`.

**Comment by luo-yuyuan** [comment]
I have started #445 and #446 as two separate PRs, so I guess I'll close this one off now :)

**Pull request closed without merging**